src/mixins/onPageLoadScrollToFragment.js

```diff
@@ -8,13 +8,57 @@
  * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
+import AppStore from 'docc-render/stores/AppStore';
+import ImageLoadingStrategy from 'docc-render/constants/ImageLoadingStrategy';
 import scrollToElement from 'docc-render/mixins/scrollToElement';
 
+function waitForImageToLoad(img) {
+  return new Promise((resolve, reject) => {
+    if (img.complete) {
+      resolve();
+    } else {
+      img.addEventListener('load', resolve, { once: true });
+      img.addEventListener('error', reject, { once: true });
+    }
+  });
+}
+
+function waitForImagesToLoad() {
+  return Promise.allSettled(
+    [...document.getElementsByTagName('img')].map(waitForImageToLoad),
+  );
+}
+
 export default {
   mixins: [scrollToElement],
   mounted() {
-    if (this.$route.hash) {
-      this.scrollToElement(this.$route.hash);
-    }
+    this.scrollToElementIfAnchorPresent();
+  },
+  updated() {
+    this.scrollToElementIfAnchorPresent();
+  },
+  methods: {
+    async scrollToElementIfAnchorPresent() {
+      const { hash } = this.$route;
+      if (!hash) {
+        return;
+      }
+
+      // Use "eager" loading for all images since they need to be loaded so
+      // that any dynamic height adjustments can be accounted for
+      const { imageLoadingStrategy } = AppStore.state;
+      AppStore.setImageLoadingStrategy(ImageLoadingStrategy.eager);
```

Comment:
would changing the loading mid render actually make a difference?

Reply:
I'll do some more testing to make sure, but I think it was necessary. Without it, some images might not load at all until they're visible due to the default lazy loading strategy, and this code would get stuck waiting forever.

Reply:
(Yeah, as I expected, it doesn't seem to work for me when commenting out the code related to toggling the loading strategy.)

Reply:
thanks for looking into it